testsuite/cluster-test/src/health/fullnode_check.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::prometheus::Prometheus;
+use crate::util::unix_timestamp_now;
+use crate::{
+    cluster::Cluster,
+    health::{HealthCheck, HealthCheckContext, ValidatorEvent},
+};
+use anyhow::{format_err, Result};
+use once_cell::sync::Lazy;
+use regex::Regex;
+use std::collections::HashMap;
+use std::time::Duration;
+
+const APP_LIBRA_VALIDATOR: &str = "libra-validator";
+const APP_LIBRA_FULLNODE: &str = "libra-fullnode";
+static VAL_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"val-(\d+)").unwrap());
+static FULLNODE_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"fn-(\d+)").unwrap());
+
+pub struct FullNodeHealthCheck {
+    cluster: Cluster,
+    prometheus: Option<Prometheus>,
+}
+
+impl FullNodeHealthCheck {
+    pub fn new(cluster: Cluster, prometheus: Option<Prometheus>) -> FullNodeHealthCheck {
+        Self {
+            cluster,
+            prometheus,
+        }
+    }
+
+    pub fn state_sync_committed_version(
+        prometheus: &Prometheus,
```

Comment:
This function could just take `&self` and get prometheus from there?

Reply:
yeah... forgot self here